lib/Models/createCatalogItemFromFileOrUrl.js

```diff
@@ -156,6 +156,7 @@ function loadItem(newCatalogItem, name, fileOrUrl) {
     newCatalogItem.data = fileOrUrl;
     newCatalogItem.dataSourceUrl = fileOrUrl.name;
     newCatalogItem.dataUrlType = "local";
+    newCatalogItem.name = fileOrUrl.name;
```

Comment:
Does this work if `fileOrUrl` is a URL? I wouldn't think it would have a name property.

Reply:
This change only effects files - the [if/else statement above](https://github.com/TerriaJS/terriajs/pull/3584/files/44c73e6560d91f41a3f5d277014489c0095007e7#diff-45581fa26beec87f4e76b259504095bcL153) basically means this block only gets applied to files.